### PR TITLE
Fix/ruby 2805 postcode validation

### PR DESCRIPTION
**Pull request by PaulDoyle-DEFRA**
Change to use the new postcode validator from defra_ruby_validators
https://eaflood.atlassian.net/browse/RUBY-2805

### CODE REVIEW

File: flood_risk_engine.gemspec [hunk 1]
```diff
@@ -36,6 +36,7 @@ Gem::Specification.new do |s|
   # which details of the last email sent by the app can be accessed
   s.add_dependency "defra_ruby_email", "~> 1"
   s.add_dependency "defra_ruby_validators"
+  s.add_dependency "faraday-retry"
```
Comment (brujeo) [inline flood_risk_engine.gemspec:39]: Shouldn't this be a Gemfile dependency?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -53,12 +53,12 @@
...
 
   # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
-  config.fixture_path = Rails.root.join("/spec/fixtures")
+  # config.fixture_path = Rails.root.join("/spec/fixtures")
 
   # If you're not using ActiveRecord, or you'd prefer not to run each of your
   # examples within a transaction, remove the following line or assign false
   # instead of true.
-  config.use_transactional_fixtures = true
+  config.use_transactional_fixtures = false
```
Comment (brujeo) [inline spec/rails_helper.rb:61]: Just wondering what the motivation was for setting this to false? As far as I'm aware transactional fixtures are way faster than the use of DB cleaner after each test

File: flood_risk_engine.gemspec (hunk 1 above)
Comment (PaulDoyle-DEFRA) [reply flood_risk_engine.gemspec:39]: Perhaps. Do you mean just for :development?

Comment (brujeo) [reply flood_risk_engine.gemspec:39]: in my understanding, faraday is only used by changelog gem, which isn't used in the runtime

File: spec/rails_helper.rb (hunk 2 above)
Comment (PaulDoyle-DEFRA) [reply spec/rails_helper.rb]: As discussed separately - I found the cause of the test failures and have pushed a commit with a fix, and reinstated config.use_transactional_fixtures = true

**Review by brujeo — APPROVED** [review]

**Pull request merged**